flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -1183,47 +1220,48 @@ public String toString() {
 	 */
 	private static class TaskCanceler implements Runnable {
 
-		private final Logger logger;
 		private final AbstractInvokable invokable;
 		private final Thread executer;
-		private final String taskName;
 		private final long taskCancellationIntervalMillis;
 		private final ResultPartition[] producedPartitions;
 		private final SingleInputGate[] inputGates;
+		private final CountDownLatch watchDogLatch;
 
 		public TaskCanceler(
-				Logger logger,
```

Comment:
I think the logger was added to prevent that the compiler needs to inject static bridge methods to access it. Why not keep it?